frappe/public/js/frappe/doctype/index.js

```diff
@@ -166,6 +206,9 @@ frappe.model.DocTypeController = class DocTypeController extends frappe.ui.form.
 				label: `${df.options} (${df.fieldname})`,
 				value: df.fieldname,
 			}));
+
+		this.validate_fetch_from(frm.doc, doctype, docname);
```

Comment:
@maharshivpatel business logic should always be server side.  

Another example: https://github.com/frappe/frappe/issues/24285

Reply:
i can also add on server side. but having it on client side gives user faster feedback ( at the same time they made mistake) and is better for ux. 

example: edited 5 fields and try to save it, error can be in any one of them.